packages/react-components/src/components/ImageGallery.tsx

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  DefaultButton,
+  IModalStyleProps,
+  IModalStyles,
+  IOverlayStyleProps,
+  IOverlayStyles,
+  IStyle,
+  IStyleFunctionOrObject,
+  Icon,
+  IconButton,
+  Layer,
+  Modal,
+  Stack,
+  mergeStyles
+} from '@fluentui/react';
+
+import React from 'react';
+import { BaseCustomStyles } from '../types';
+import {
+  cancelIcon,
+  closeButtonStyles,
+  controlBarContainerStyle,
+  downloadButtonStyle,
+  downloadIcon,
+  downloadIconStyle,
+  focusTrapZoneStyle,
+  headerStyle,
+  imageContainer,
+  imageStyle,
+  overlayStyles,
+  scrollableContentStyle,
+  smallDownloadButtonContainerStyle,
+  titleBarContainerStyle,
+  titleStyle
+} from './styles/ImageGallery.style';
+import { useTheme } from '../theming/FluentThemeProvider';
+import { isDarkThemed } from '../theming/themeUtils';
+
+/**
+ * Fluent styles for {@link ImageGallery}.
+ *
+ * @beta
+ */
+export interface ImageGalleryStylesProps extends BaseCustomStyles {
+  /** Styles for the ImageGallery Modal. */
+  modal?: IStyleFunctionOrObject<IModalStyleProps, IModalStyles>;
+  /** Styles for the ImageGallery Modal overlay. */
+  overlay?: IStyleFunctionOrObject<IOverlayStyleProps, IOverlayStyles>;
+  /** Styles for the ImageGallery header bar. */
+  header?: IStyle;
+  /** Styles for the ImageGallery titleBar container. */
+  titleBarContainer?: IStyle;
+  /** styles for the title label */
+  title?: IStyle;
+  /** Styles for the ImageGallery controlBar container. */
+  controlBarContainer?: IStyle;
+  /** Styles for the download button. */
+  downloadButton?: IStyle;
+  /** Styles for the small download button when screen width is smaller than 25 rem. */
+  smallDownloadButton?: IStyle;
+  /** Styles for the close modal icon. */
+  closeIcon?: IStyle;
+  /** Styles for the image. */
+  image?: IStyle;
+}
+
+/**
+ * Props for {@link ImageGallery}.
+ *
+ * @beta
+ */
+export interface ImageGalleryImageProps {
+  imageUrl: string;
+  downloadFileName: string;
```

Comment:
Can this just be called fileName? (same with other references for `downloadFileName`)

Reply:
> Can this just be called fileName? (same with other references for `downloadFileName`)

Would it be less clear to Contosos what fileName is used for?

Reply:
This is the file name developers will define for what the saved file should be right? Maybe we can name it as much `saveasFilename` or something along those lines